torch/_refs/__init__.py

```diff
@@ -804,7 +804,7 @@ def logsumexp(
         dim = (dim,)
     if self.numel() == 0:
         return torch.sum(torch.exp(self), dim, keepdim).log()
-    maxes = torch.amax(self, dim, keepdim=True)
+    maxes = torch.amax(real(self), dim, keepdim=True)
```

Comment:
Please call `torch.real` (and in general, don't call other decompositions directly), so that the call goes through the dispatcher rather than always being inlined.